bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/LegacyHierarchicalLedgerManagerFactory.java

```diff
@@ -79,7 +79,7 @@ public LedgerIdGenerator newLedgerIdGenerator() {
 
     @Override
     public LedgerManager newLedgerManager() {
-        return new HierarchicalLedgerManager(conf, zk);
```

Comment:
oh, the original code seems weird, why it return HierarchicalLedgerManager?

Reply:
yes, I do see few other issues with 057af8dbce6c08794eb8b46ca52ca13f222d9bbb . Tried to address in this commit and we noticed other issues as well - Issue #1014